File: ruby/muse/algorithms/sorting/SelectionSortTest.rb
require "muse/algorithms/sorting/SelectionSort"
require "muse/util/SequenceBuilder"
require "muse/util/Sequences"
require "muse/util/TestRunner"

def testSelectionSort()
    size = 32768

    arr = Array.new(size)
    SequenceBuilder.packRandom(arr)

    checksum = Sequences.parityChecksum(arr)

    SelectionSort.sort(arr)

    if Sequences.parityChecksum(arr) != checksum
        return false
    end

    Sequences.isSorted(arr)
end

if __FILE__ == $PROGRAM_NAME
    TestRunner.pick(testSelectionSort())
end

File: python/muse/algorithms/sorting/SelectionSortTest.py
from muse.algorithms.sorting import SelectionSort
from muse.util import SequenceBuilder, Sequences, TestRunner


def testSelectionSort() -> bool:
    size = 32768

    arr = [0] * size
    SequenceBuilder.packRandom(arr)

    checksum = Sequences.parityChecksum(arr)

    SelectionSort.sort(arr)

    if Sequences.parityChecksum(arr) != checksum:
        return False

    return Sequences.isSorted(arr)


def main() -> None:
    TestRunner.pick(testSelectionSort)


if __name__ == "__main__":
    main()
